src/virtual-machine.js

```diff
@@ -177,34 +177,38 @@ class VirtualMachine extends EventEmitter {
     }
 
     /**
-     * Load a project from a Scratch 2.0 JSON representation.
-     * @param {?string} json JSON string representing the project.
+     * Load a Scratch project from a .sb, .sb2, .sb3 or json string.
+     * @param {string | object} input A json string, object, or ArrayBuffer representing the project to load.
      * @return {!Promise} Promise that resolves after targets are installed.
      */
-    loadProject (json) {
-        // @todo: Handle other formats, e.g., Scratch 1.4, Scratch 3.0.
-        return this.fromJSON(json);
-    }
+    loadProject (input) {
+        if (typeof input === 'object' && !(input instanceof ArrayBuffer)) {
+            // validate expects a string or buffer as input
+            input = JSON.stringify(input);
+        }
 
-    /**
-     * Load a project from a Scratch 3.0 sb3 file containing a project json
-     * and all of the sound and costume files.
-     * @param {Buffer} inputBuffer A buffer representing the project to load.
-     * @return {!Promise} Promise that resolves after targets are installed.
-     */
-    loadProjectLocal (inputBuffer) {
-        // TODO need to handle sb2 files as well, and will possibly merge w/
-        // above function
-        return JSZip.loadAsync(inputBuffer)
-            .then(sb3File => {
-                sb3File.file('project.json').async('string')
-                    .then(json => {
-                        // TODO error handling for unpacking zip/not finding project.json
-                        json = JSON.parse(json); // TODO catch errors here (validation)
-                        return sb3.deserialize(json, this.runtime, sb3File)
-                            .then(({targets, extensions}) =>
-                                this.installTargets(targets, extensions, true));
-                    });
+        // Clear the current runtime
+        this.clear();
+
+        const validationPromise = new Promise((resolve, reject) => {
+            validate(input, (error, res) => {
+                if (error) {
+                    reject(error);
+                }
+                if (res) {
+                    resolve(res);
+                }
+            });
+        });
```

Comment:
If we end up doing this Node-style callback <=> Promise conversion more often it might make sense to make a utility function or two for it. I think this is the first one I've seen in VM, though, so I think this is fine for now.